Client/Frontend/Notifications/NotificationRootViewController.swift

```diff
@@ -159,9 +161,34 @@ private extension NotificationRootViewController {
         }
     }
 
-    @objc func didFinishSyncing() {
-        showingNotification = false
-        dispatch_async(dispatch_get_main_queue()) {
+    @objc func didFinishSyncing(notification: NSNotification) {
+        let syncMessage: NSAttributedString?
+        if let syncManager = notification.object as? SyncManager,
+            let syncDisplayState = syncManager.syncDisplayState {
+            switch syncDisplayState {
+            case .Bad(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowErrorTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            case .Stale(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowWarningTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            default: syncMessage = nil
+            }
+        } else {
+            syncMessage = nil
+        }
+
+        guard let displayMessage = syncMessage else {
+            showingNotification = false
+            return dispatch_async(dispatch_get_main_queue()) {
+                self.hideStatusNotification()
+            }
+        }
+        
+        notificationView.titleLabel.attributedText = displayMessage
+        notificationView.ellipsisLabel.hidden = true
+        self.notificationView.endAnimation()
+        let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(4 * Double(NSEC_PER_SEC)))
+        dispatch_after(delayTime, dispatch_get_main_queue()) {
+            self.showingNotification = false
```

Comment:
> We can't attach the syncDisplayResult as NSNotification only allows us to attach an object. I thought about only attaching the message, but then I lose the information about whether the message is .Bad or .Stale for colouring. I'll try and look at ways of wrapping that enum value as an object to pass instead.

Argh Objective-C pains. Between the option of wrapping it with some kind of object vs passing in the `syncManager` I'm not sure what to do. I guess my fear with passing the `syncManager` is observers retaining a reference to it which could lead to confusing memory management and retain cycles. Or maybe it's a non-issue?